### PR TITLE
Add connection to HANA

### PR DESCRIPTION
**Pull request by lucboj**
# Description

Added the basics for connecting to a HANA instance. Can run "task mongosh" without error and connect to `hana` handler.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/hana/hanadb/pool.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// limitations under the License.
+
+package hanadb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "SAP/go-hdb/driver" //nolint:goimports // goimports and gci are conflicting
```
Comment (AlekSi) [inline internal/handlers/hana/hanadb/pool.go:22]: It should imported as `github.com/SAP/go-hdb/driver`

File: .gitignore [hunk 2]
```diff
@@ -22,3 +22,6 @@ website/build/
 
 # for local Docker Compse changes; see https://docs.docker.com/compose/extends/#multiple-compose-files
 docker-compose.override.yml
+
+# enviroment variable files like dotenv
+.env
```
Comment (AlekSi) [inline .gitignore:27]: See the comment on top of the file

File: go.mod [hunk 3]
```diff
@@ -4,7 +4,6 @@ go 1.20
 
 require (
 	github.com/AlekSi/pointer v1.2.0
-	github.com/SAP/go-hdb v1.1.1
```
Comment (AlekSi) [inline go.mod:7]: That one is needed

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (lucboj) [reply go.mod:7]: This depends on which Go driver we would want to use. There is the one on Github and then there is the official HANA Go driver which can be found here https://tools.hana.ondemand.com/#hanatools. I have created now a script and figured how to set it up for Github Actions. 

I do not know the difference. So, I guess up to you whether it should be the Github or the official driver which we use.

File: .gitignore (hunk 2 above)
Comment (lucboj) [reply .gitignore:27]: If you mean "dotenv" then that is how the file is called. But I can of course also write .env.

File: internal/handlers/hana/hanadb/pool.go (hunk 1 above)
Comment (lucboj) [reply internal/handlers/hana/hanadb/pool.go:22]: See other comment

File: go.mod (hunk 3 above)
Comment (AlekSi) [reply go.mod:7]: Well, currently, CI fails due to missing dependency

Comment (lucboj) [reply go.mod:7]: Yes, I also did not add the script and the changes yet. More or less just found the solution 30 minutes ago :D 
But if we decide for the official driver, I can add it.

**Comment by lucboj** [comment]
@AlekSi I have added the script and changed the workflows, so that the CI is not failing. The current failures I think are unrelated to my changes.

File: go.mod (hunk 3 above)
Comment (AlekSi) [reply go.mod:7]: Please see Slack

File: .gitignore (hunk 2 above)
Comment (AlekSi) [reply .gitignore:27]: I mean this comment:

https://github.com/FerretDB/FerretDB/blob/a0cd2e5c5bf56cc51747f399189d924ee31f8197/.gitignore#L1-L2

That line should be in your global `.gitignore` file, not there.

Please also do not resolve conversations after responding:
https://github.com/FerretDB/FerretDB/blob/a0cd2e5c5bf56cc51747f399189d924ee31f8197/.github/PROCESS.md?plain=1#L83-L85

(but please read the whole file)

File: cmd/ferretdb/main.go [hunk 4]
```diff
@@ -71,6 +71,8 @@ var cli struct {
 
 	PostgreSQLURL string `name:"postgresql-url" default:"${default_postgresql_url}" help:"PostgreSQL URL for 'pg' handler."`
 
+	HANAURL string `name:"hana-url" help:"SAP HANA URL for 'hana' handler"`
```
[the remark this answers is not in the document]
Comment (lucboj) [reply cmd/ferretdb/main.go]: I hope I did this correctly

File: .gitignore (hunk 2 above)
Comment (AlekSi) [reply .gitignore:27]: Well, one could add them to dotenv's `.envrc` and have the same problem. But we are not trying to cover all tools contributors could use. Project's `.gitignore` file is for project-specific ignore patterns.

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -192,6 +192,8 @@ jobs:
         run: bin/task env-setup
         working-directory: source
 
+
+
```
Comment (AlekSi) [inline .github/workflows/go.yml:196]: Please revert this change

File: Taskfile.yml [hunk 6]
```diff
@@ -273,6 +273,19 @@ tasks:
         --tigris-url=127.0.0.1:8081
         --test-records-dir=tmp/records
 
+  run-hana:
+    deps: [build-local]
+    dotenv: ['.env', '{{.ENV}}/.env.', '{{.HOME}}/.env'] # Create a .env file in root folder with: export hana_url=”insert-url-here”
```
Comment (AlekSi) [inline Taskfile.yml:278]: We recommend contributors to use direnv, see https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md. But we also should not rely on particular tools and files they use.

In this case, FerretDB will handle `FERRETDB_HANA_URL` environment variable automatically, so there is no need to pass `--hana-url` flag.

File: go.mod [hunk 7]
```diff
@@ -18,12 +17,14 @@ require (
...
-	golang.org/x/exp v0.0.0-20230224173230-c95f2b4c22f2
-	golang.org/x/net v0.7.0
-	golang.org/x/sys v0.5.0
+	golang.org/x/crypto v0.7.0 // indirect; always use @latest
+	golang.org/x/exp v0.0.0-20230304125523-9ff063c70017
+	golang.org/x/net v0.8.0
+	golang.org/x/sys v0.6.0
 )
 
+require github.com/SAP/go-hdb v1.1.1
```
Comment (AlekSi) [inline go.mod:26]: Please move it to the group above, where it was

**Review by AlekSi — COMMENTED** [review]
> I marked all done items in this checklist.

Please check all items you did in the PR checklist

File: Taskfile.yml (hunk 6 above)
Comment (lucboj) [reply Taskfile.yml:278]: Ahh okay, super! I was not aware of this being possible. How do you do this? I cannot find FERRETDB_HANA_URL any place in the code? Just asking out of interest 😊

Comment (AlekSi) [reply Taskfile.yml]: `kong.DefaultEnvars` does that: https://pkg.go.dev/github.com/alecthomas/kong#DefaultEnvars

**Review by AlekSi — APPROVED** [review]

File: Taskfile.yml (hunk 6 above)
Comment (lucboj) [reply Taskfile.yml]: Thank you!
